app/controllers/foreign/income/ForeignPropertyRentalIncomeController.scala

```diff
@@ -30,44 +30,55 @@ import views.html.foreign.income.ForeignPropertyRentalIncomeView
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  navigator: ForeignPropertyNavigator,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: ForeignPropertyRentalIncomeFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: ForeignPropertyRentalIncomeView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController with I18nSupport {
 
-
-
-  def onPageLoad(taxYear: Int,  mode: Mode,countryCode:String): Action[AnyContent] = (identify andThen getData andThen requireData) {
-    implicit request =>
+  def onPageLoad(taxYear: Int, mode: Mode, countryCode: String): Action[AnyContent] =
+    (identify andThen getData andThen requireData) { implicit request =>
       val form = formProvider(request.user.isAgentMessageKey)
       val preparedForm = request.userAnswers.get(ForeignPropertyRentalIncomePage(countryCode)) match {
-        case None => form
+        case None        => form
         case Some(value) => form.fill(value)
       }
 
       Ok(view(preparedForm, taxYear, request.user.isAgentMessageKey, mode, countryCode))
-  }
+    }
 
-  def onSubmit(taxYear: Int, mode: Mode, countryCode:String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
-    implicit request =>
+  def onSubmit(taxYear: Int, mode: Mode, countryCode: String): Action[AnyContent] =
+    (identify andThen getData andThen requireData).async { implicit request =>
       val form = formProvider(request.user.isAgentMessageKey)
-      form.bindFromRequest().fold(
-        formWithErrors =>
-          Future.successful(BadRequest(view(formWithErrors, taxYear, request.user.isAgentMessageKey, mode,countryCode))),
-
-        value =>
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(ForeignPropertyRentalIncomePage(countryCode), value))
-            _              <- sessionRepository.set(updatedAnswers)
-          } yield Redirect(navigator.nextPage(ForeignPropertyRentalIncomePage(countryCode), taxYear, mode, request.userAnswers, updatedAnswers))
-      )
-  }
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors =>
+            Future
+              .successful(BadRequest(view(formWithErrors, taxYear, request.user.isAgentMessageKey, mode, countryCode))),
+          value =>
+            for {
+              updatedAnswers <-
+                Future.fromTry(request.userAnswers.set(ForeignPropertyRentalIncomePage(countryCode), value))
+              updatedAnswersWithCountryCode <-
+                Future.fromTry(updatedAnswers.set(ForeignIncomeSectionAddCountryCode(countryCode), countryCode))
+              _ <- sessionRepository.set(updatedAnswersWithCountryCode)
+            } yield Redirect(
+              navigator.nextPage(
+                ForeignPropertyRentalIncomePage(countryCode),
+                taxYear,
+                mode,
```

Comment:
status still showing in progress when completed 

<img width="439" alt="Screenshot 2024-12-06 at 11 23 26" src="https://github.com/user-attachments/assets/96b9ecd9-f731-4494-9afd-9c327bbaec9f">